### PR TITLE
Proposed fix for #DCC8499

### PR DESCRIPTION
**Pull request by diego-ninja**
Proposed fix for issue commented in #8499.

### CODE REVIEW

**Comment by greg0ire** [comment]
Since this is a fix, you should target 2.8.x
Out of curiosity, why did you target `master`?

File: lib/Doctrine/ORM/UnitOfWork.php [hunk 1]
```diff
@@ -1970,7 +1970,12 @@ public function lock($entity, $lockMode, $lockVersion = null)
 
                 $entityVersion = $class->versionProperty->getValue($entity);
 
-                if ($entityVersion !== $lockVersion) {
+                if ($lockVersion instanceof \DateTime && $entityVersion instanceof \DateTime) {
```
Comment (Warxcell) [inline lib/Doctrine/ORM/UnitOfWork.php]: What if it's DateTimeImmutable?

Comment (beberlei) [reply lib/Doctrine/ORM/UnitOfWork.php]: Yes, must be 
```suggestion
                if ($lockVersion instanceof \DateTimeInterface && $entityVersion instanceof \DateTimeInterface) {
```

**Comment by beberlei** [comment]
Fix is correct, but it requires tests.

**Comment by diego-ninja** [comment]
> Since this is a fix, you should target 2.8.x
> Out of curiosity, why did you target `master`?



> Since this is a fix, you should target 2.8.x
> Out of curiosity, why did you target `master`?

Following the guidelines in CONTRIBUTING.md that states only PR to master will be accepted. Sorry if I did it wrong.

**Comment by greg0ire** [comment]
> Following the guidelines in CONTRIBUTING.md that states only PR to master will be accepted. Sorry if I did it wrong.

Oh my! We're the ones that did something wrong apparently, let me PR that real quick!

**Comment by greg0ire** [comment]
See https://github.com/doctrine/orm/pull/8502

**Comment by Warxcell** [comment]
https://github.com/doctrine/orm/blob/2.8.x/lib/Doctrine/ORM/OptimisticLockException.php#L73-L74

Same must be done here.

**Comment by greg0ire** [comment]
You will have to rebase onto `2.8.x` : `git rebase --onto origin/2.8.x origin/master master`. If that sounds confusing, that's because naming your branch like the target branch was not a great idea :stuck_out_tongue:

**Comment by diego-ninja** [comment]
> git rebase --onto origin/2.8.x origin/master master

I'm going to close this PR and create a new one (to 2.8.x branch) with the fix, the change in the exception and some tests.

**Comment by diego-ninja** [comment]
Okie, closing this PR and start a clean one from 2.8.x branch. Sorry for the misunderstandings.

**Pull request closed without merging**